File: src/org/cappuccino_project/ide/intellij/plugin/stubs/interfaces/ObjJImplementationStub.java
package org.cappuccino_project.ide.intellij.plugin.stubs.interfaces;

import org.cappuccino_project.ide.intellij.plugin.psi.impl.ObjJImplementationDeclarationImpl;

import javax.annotation.Nullable;

public interface ObjJImplementationStub extends ObjJClassDeclarationStub<ObjJImplementationDeclarationImpl> {

    @Nullable
    String getSuperClassName();
    @Nullable
    String getCategoryName();
    boolean isCategory();

}

File: src/org/cappuccino_project/ide/intellij/plugin/stubs/interfaces/ObjJImplementationStub.kt
package org.cappuccino_project.ide.intellij.plugin.stubs.interfaces

import org.cappuccino_project.ide.intellij.plugin.psi.impl.ObjJImplementationDeclarationImpl

interface ObjJImplementationStub : ObjJClassDeclarationStub<ObjJImplementationDeclarationImpl> {

    val superClassName: String?
    val categoryName: String?
    val isCategory: Boolean

}
